exec/java-exec/src/test/java/org/apache/drill/exec/store/bson/TestBsonRecordReader.java

```diff
@@ -49,17 +50,20 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class TestBsonRecordReader extends BaseTestQuery {
+public class TestBsonRecordReader {
+  private static BufferAllocator allocator;
```

Comment:
What is a reason to define `BufferAllocator` and other fields as static? What if tests are executed in parallel?

Reply:
JUnit requires tests annotated with @BeforeClass and @AfterClass to be static so these variables have to be static. Currently the Drill unit tests do not support concurrent execution of methods within a test class within the same process. Drill has surefire launch multiple test processes and the the test classes to execute are divided among the test processes. Within each forked surefire process tests are executed sequentially, so this will not be an issue.

Reply:
What is a reason `allocator` and other variables are initialized in `@BeforeClass`? Can they be initialized in `@Before` instead?